### PR TITLE
Require fewer secrets for local dev

### PR DESCRIPTION
**Pull request by mxmeinhold**
This PR gets most of the way to enabling secretless local development, but not quite. 

I haven't wrapped SSO to enable developing completely offline, but this PR adds:
- Local postgresql instructions for running in a container
- LDAPWrapper to allow mocking accounts and decouple from ldap
- Prevents errors by disabling onesignal if secrets aren't configured
- Expands the logging on initialisation a little to show more of what's running locally

SSO will need a wrapper or overrides for the core methods, similar to LDAPWrapper, which is beyond what I'm prepared to do at the moment.

This should make hacktoberfest somewhat simpler, as well as making it easier to work in codespaces (though this is still heavily limited by sso).

### CODE REVIEW

**Review by devinmatte — APPROVED** [review]
This looks great! We should do the same on a few other projects to enable easier local dev after we've graduated

**Comment by mxmeinhold** [comment]
That's the plan, once I remember to get around to it.

File: README.md [hunk 1]
```diff
@@ -47,6 +47,20 @@ If it doesn't work for some reason, you may have to globally install gulp throug
 npm install -g gulp
 ```
 
+### Local Development
+* PostgreSQL
+You'll need a postgres instance to use as a development DB.
+You can use an existing database, like the instance used for the dev branch, use a database on another server, or spin up a container using docker or podman. 
+To get setup using docker, run
+```bash
+docker run --name packet-postgres -e POSTGRES_PASSWORD=mysecretpassword -d -p 5432:5432 postgres
```
Comment (mbillow) [inline README.md:56]: Any reason to not just toss this into a docker-compose file, [like I did for self-service](https://github.com/ComputerScienceHouse/self-service/blob/main/docker-compose.yaml)?

File: README.md [hunk 2]
```diff
@@ -47,6 +47,20 @@ If it doesn't work for some reason, you may have to globally install gulp throug
...
+```
+After the container starts up, you should be able to connect with the connection string `postgresql://postgres:mysecretpassword@localhost:5432/postgres`, which is the default connection string in `config.env.py`.
+Once the container is up, run the following to set up the database tables.
+```bash
+flask db upgrade
```
Comment (mbillow) [inline README.md:61]: We should really get in the habit of having the application itself run migrations automatically when it starts. I'm not sure there is any reason not to.

File: README.md (hunk 1 above)
Comment (mxmeinhold) [reply README.md:56]: I use podman, so docker-compose is not something I default to yet.

**Pull request merged**